storage/utils.go

```diff
@@ -2,28 +2,115 @@ package storage
 
 import (
 	"context"
+	"fmt"
+	"github.com/Peripli/service-manager/pkg/instance_sharing"
+	"github.com/Peripli/service-manager/pkg/log"
 	"github.com/Peripli/service-manager/pkg/query"
 	"github.com/Peripli/service-manager/pkg/types"
 	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/tidwall/gjson"
 )
 
 func GetServiceOfferingByServiceInstanceId(repository Repository, ctx context.Context, serviceInstanceId string) (*types.ServiceOffering, error) {
 	byID := query.ByField(query.EqualsOperator, "id", serviceInstanceId)
 	criteria := query.CriteriaForContext(ctx)
 	obj, err := repository.Get(ctx, types.ServiceInstanceType, append(criteria, byID)...)
 	if err != nil {
-		return nil, util.HandleStorageError(err, types.ServiceInstanceType.String())
+		log.C(ctx).Errorf("Failed retrieving the service instance by ID: %s", serviceInstanceId)
+		return nil, util.HandleStorageError(util.ErrNotFoundInStorage, types.ServiceInstanceType.String())
 	}
 	serviceInstance := obj.(*types.ServiceInstance)
 	planObject, err := repository.Get(ctx, types.ServicePlanType, query.ByField(query.EqualsOperator, "id", serviceInstance.ServicePlanID))
 	if err != nil {
-		return nil, util.HandleStorageError(err, types.ServicePlanType.String())
+		log.C(ctx).Errorf("Failed retrieving the service plan by ID: %s", serviceInstance.ServicePlanID)
+		return nil, util.HandleStorageError(util.ErrNotFoundInStorage, types.ServiceInstanceType.String())
 	}
 	plan := planObject.(*types.ServicePlan)
 	serviceObject, err := repository.Get(ctx, types.ServiceOfferingType, query.ByField(query.EqualsOperator, "id", plan.ServiceOfferingID))
 	if err != nil {
-		return nil, util.HandleStorageError(err, types.ServiceOfferingType.String())
+		log.C(ctx).Errorf("Failed retrieving the service offering by ID: %s", plan.ServiceOfferingID)
+		return nil, util.HandleStorageError(util.ErrNotFoundInStorage, types.ServiceInstanceType.String())
 	}
 	service := serviceObject.(*types.ServiceOffering)
 	return service, nil
 }
+
+func GetObjectByField(ctx context.Context, repository Repository, objectType types.ObjectType, byKey, byValue string) (types.Object, error) {
+	byID := query.ByField(query.EqualsOperator, byKey, byValue)
+	dbObject, err := repository.Get(ctx, objectType, byID)
+	if err != nil {
+		log.C(ctx).Errorf("GetObjectByField failed retrieving the %s by %s: %s", objectType.String(), byKey, byValue)
+		return nil, util.HandleStorageError(util.ErrNotFoundInStorage, objectType.String())
+	}
+	return dbObject, nil
+}
+
+func IsReferencePlan(ctx context.Context, repository Repository, objectType, byKey, servicePlanID string) (bool, error) {
```

Comment:
isn't `byKey` always `id`? if so, don't accept it as parameter

Reply:
can be by "catalog_id" as well.